presto-main/src/test/java/com/facebook/presto/block/TestArrayBlock.java

```diff
@@ -0,0 +1,245 @@
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder intermediateBlockBuilder = new ArrayBlockBuilder(BIGINT, new BlockBuilderStatus(), 100, 100);
+                for (int j = 0; j < expectedValues[i].length; j++) {
+                    if (expectedValues[i][j] == null) {
+                        intermediateBlockBuilder.appendNull();
+                    }
+                    else {
+                        BlockBuilder innerMostBlockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                        for (long v : expectedValues[i][j]) {
+                            BIGINT.writeLong(innerMostBlockBuilder, v);
+                        }
+                        intermediateBlockBuilder.writeObject(innerMostBlockBuilder.build()).closeEntry();
+                    }
+                }
+                blockBuilder.writeObject(intermediateBlockBuilder.build()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    private static void assertArrayValues(long[][] expectedValues)
+    {
+        BlockBuilder blockBuilder = new ArrayBlockBuilder(BIGINT, new BlockBuilderStatus(), 100, 100);
+        for (int i = 0; i < expectedValues.length; i++) {
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBlockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                for (long v : expectedValues[i]) {
+                    BIGINT.writeLong(elementBlockBuilder, v);
+                }
+                blockBuilder.writeObject(elementBlockBuilder).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    private static void assertArrayValues(Slice[][] expectedValues)
+    {
+        BlockBuilder blockBuilder = new ArrayBlockBuilder(VARCHAR, new BlockBuilderStatus(), 100, 100);
+        for (int i = 0; i < expectedValues.length; i++) {
+            if (expectedValues[i] == null) {
+                blockBuilder.appendNull();
+            }
+            else {
+                BlockBuilder elementBlockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), expectedValues[i].length);
+                for (Slice v : expectedValues[i]) {
+                    VARCHAR.writeSlice(elementBlockBuilder, v);
+                }
+                blockBuilder.writeObject(elementBlockBuilder.build()).closeEntry();
+            }
+        }
+        assertBlock(blockBuilder, expectedValues);
+        assertBlock(blockBuilder.build(), expectedValues);
+    }
+
+    private static Object[] alternatingNullValues(Object[] objects)
```

Comment:
there seems to be a lot of overlap with `AbstractTestBlock` I think you can remove most of it